### PR TITLE
Add default enterprise suffix when create an enterprise method (#958)

### PR DESCRIPTION
**Pull request by Daoq**
Add more descriptions help users to utilize the method and default suffix format to prevent from the 406 status code.

### CODE REVIEW

File: github/github.go [hunk 1]
```diff
@@ -278,13 +278,16 @@ func NewClient(httpClient *http.Client) *Client {
 }
 
 // NewEnterpriseClient returns a new GitHub API client with provided
-// base URL and upload URL (often the same URL).
-// If either URL does not have a trailing slash, one is added automatically.
+// base URL and upload URL (often the same URL and is your enterprise github host name).
```
[the remark this answers is not in the document]
Comment (Daoq) [reply github/github.go]: Much better! Thanks @gmlewis

**Comment by Daoq** [comment]
> I'm LGTM on these changes, but is there a bug or GitHub API docs explaining the usage pattern of the api/v3/ pattern? Just want to see if this is the best approach over returning an error.

The GitHub Enterprise Server 2.16 Documentation pointed out the users, by default, should follow the format like http(s)://[hostname]/api/v3

https://developer.github.com/enterprise/2.16/v3/

**Comment by gmlewis** [comment]
Hi @Daoq, I'm waiting for a couple minor changes if you don't mind. Please see above.
Thanks.

File: github/github.go [hunk 2]
```diff
@@ -278,13 +278,16 @@ func NewClient(httpClient *http.Client) *Client {
...
+// If either URL does not have a trailing slash or the suffix api/v3, one is added automatically.
```
[the remark this answers is not in the document]
Comment (Daoq) [reply github/github.go]: LGTM

File: github/github.go [hunk 3]
```diff
@@ -278,13 +278,16 @@ func NewClient(httpClient *http.Client) *Client {
...
 // If a nil httpClient is provided, http.DefaultClient will be used.
 //
 // Note that NewEnterpriseClient is a convenience helper only;
 // its behavior is equivalent to using NewClient, followed by setting
 // the BaseURL and UploadURL fields.
+//
+// Another important thing is that by default, the github enterprise url format
```
[the remark this answers is not in the document]
Comment (Daoq) [reply github/github.go]: LGTM

File: github/github.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gmlewis) [reply github/github.go]: This is still unresolved.

File: github/github.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (gmlewis) [reply github/github.go]: This is still unresolved.

File: github/github.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (gmlewis) [reply github/github.go]: This is still unresolved.

File: github/github.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gmlewis) [reply github/github.go]: @Daoq - this is still unresolved.

File: github/github.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (gmlewis) [reply github/github.go]: @Daoq - this is also unresolved.

File: github/github.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (gmlewis) [reply github/github.go]: @Daoq - we attempt to be very careful about the spelling of the GitHub trademark with a capital `G` and capital `H` in this repo. This request is still unresolved.

File: github/github.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Daoq) [reply github/github.go]: Sorry, I resolved. Thanks.

File: github/github.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Daoq) [reply github/github.go]: Sorry, I resolved. Thanks.

File: github/github.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Daoq) [reply github/github.go]: Sorry, I resolved. Thanks.

**Pull request merged**